core/src/main/scala/io/finch/Bootstrap.scala

```diff
@@ -81,7 +82,7 @@ class Bootstrap[F[_], ES <: HList, CTS <: HList](
     ts.apply(endpoints, opts, ctx)
   }
 
-  def toService(implicit F: Effect[F], ts: Compile[F, ES, CTS]): Service[Request, Response] =
+  def toService(implicit F: Async[F], dispatcher: Dispatcher[F], ts: Compile[F, ES, CTS]): Service[Request, Response] =
```

Comment:
I have very limited knowledge of CE3, but from what I'm reading we don't need to ask for Dispatcher[F] if we already have Async[F]. Looks like we can just summon a dispatcher: https://typelevel.org/cats-effect/docs/std/dispatcher

Reply:
I'm not very confident about this... Looking at `Dispatcher[IO]`, it seems to me that instantiation is expensive, but at the same time doesn't seem to be that much.

Looking forward, I've sturborned across a discussion between [Fabio Labella and @joroKr21 on Gitter ](https://gitter.im/typelevel/cats-effect?at=5fbebdbafe857d71dc953fe3)on the same concern, where it's stated that Dispatcher instantiation:

> isn't that heavyweight either, just a Fiber and some AtomicReferences
> it's very very lightweight in fact

Reply:
But no matter, dispatcher instanciation return a `Resource[F, Dispatcher[F]]`, which we cannot run synchronously on its own...

Reply:
I guess that's my point exactly? Shouldn't we `use` that resource instead of accessing Dispatcher directly? I think we're side-stepping the API for some reason.

This is the use method I'm referring to: https://github.com/typelevel/cats-effect/blob/series/3.x/kernel/shared/src/main/scala/cats/effect/kernel/Resource.scala#L206

Reply:
`use` also would change the signature to `IO`

Reply:
Ok, I see. I'm reading this doc (https://typelevel.org/cats-effect/docs/migration-guide) and it, at least, says that we should use Dispatcher as a type-class, b/c it's not one:

> An alternative approach to creating a Dispatcher on our own would be taking one as a parameter. It is recommended to pass it explicitly, as Dispatcher isn't a type class and there can be many instances of it in an application.

We somehow need to bubble Dispatcher construction up, I assume. Should it maybe happen as part of a Bootstrap? Like Bootstrap.toService would return you a `Resource[Service]` with `Dispatcher` fused in?

I understand we also need to figure out how to run those EndpointResult methods since they are happening outside of Bootstrap. Maybe for those we should use ask for Dispatcher explicitly and expect users to instantiate one themselves? Maybe it's not too much trouble since it's only happening in tests.

Reply:
Updated accordingly